server/pages/index.js

```diff
@@ -120,13 +120,9 @@ function getCurrentCommitShortChecksum() {
 }
 
 function getDefaultContext( request ) {
-	let initialServerState = {};
-	// We don't cache routes with query params
-	if ( isEmpty( request.query ) ) {
```

Comment:
Right, I didn't highlight this enough in the description (will do) but that's one of the reason you're marked as a reviewer here, since you have experience on SSR :)
As far as I could test, I tried every isomorphic sections and didn't see any rendering issues on the current version of calypso, I might have missed some cases though.